### PR TITLE
Generate: TF supports multiple eos tokens

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

TF generation test addition PR 4 (out of 4) 👉 All PT generation integration tests for features that also exist in TF are now framework-agnostic 🎉 

To complete this last step, I've added support to multiple EOS tokens in TF. 

Because this touches the core of the TF generation code, slow tests were run for:
- [x] TF GPT2
- [x] TF T5

### CODE REVIEW

File: src/transformers/generation/tf_utils.py [hunk 1]
```diff
@@ -1230,7 +1230,7 @@ def _prepare_input_ids_for_generation(
     ) -> tf.Tensor:
         if self.config.is_encoder_decoder and encoder_outputs is not None:
             # make dummy input_ids with value -100, as a sanity check ensuring that they won't be used for encoding
-            shape = encoder_outputs.last_hidden_state.size()[:-1]
+            shape = encoder_outputs.last_hidden_state.shape[:-1]
```
Comment (gante) [inline src/transformers/generation/tf_utils.py:1233]: The power of copy-paste: I added PT syntax in TF code in the last PR 🙃 Fortunately, the tests added in this PR caught it.

File: tests/generation/test_framework_agnostic.py [hunk 2]
```diff
@@ -486,3 +490,208 @@ def test_generate_inputs_and_encoder_kwargs(self):
...
+            "top_p": 0.7,
+            "top_k": 10,
+            "temperature": 0.7,
+        }
+        # Note: random-based tests naturally have framework-dependent results.
+        expectation = 15
+        if is_pt:
+            eos_token_id = 846
+        else:
+            eos_token_id = 319
```
Comment (gante) [inline tests/generation/test_framework_agnostic.py:645]: This is the only test where I'm not sure whether being framework-agnostic is the best pattern.

Problem: seeded sampling is framework-dependent, i.e. seed `0` does not yield the same result in the two frameworks.

The solution adopted: set a different test constant for each framework. Here, `eos_token_id` is the 15th generated token when we set seed `0`

Comment (sgugger) [reply tests/generation/test_framework_agnostic.py:645]: Might be easier to test it in the different framework then.

**Review by sgugger — APPROVED** [review]
Thanks a lot!

**Comment by gante** [comment]
The two failing tests are known failures -- merging

**Pull request merged**